skl2onnx/operator_converters/PolynomialFeatures.py

```diff
@@ -0,0 +1,70 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import numpy as np
+from ..proto import onnx_proto
+from ..common._apply_operation import apply_concat, apply_cast
+from ..common._registration import register_converter
+
+
+def convert_sklearn_polynomial_features(scope, operator, container):
+    op = operator.raw_operator
+    transformed_columns = [None] * (op.n_output_features_)
+
+    combinations = op._combinations(op.n_input_features_, op.degree,
+                                 op.interaction_only,
+                                 op.include_bias)
+
+    for i, comb in enumerate(combinations):
```

Comment:
I was imagining a dot product to produce all spuare features another one to produce feature of degree 3...

Reply:
I don't quite understand your comment. Are you asking me to explain the code? Two features, say a and b with degree 3 will produce these features: 1, a, b, a^2, ab, b^2, a^3, a^2b, ab^2, b^3. combinations will contain the element indices to consider for the current product.
() -> 1
(0,) -> a
(1,) -> b
(0, 0) -> a^2
(0, 1) -> ab
(1, 1) -> b^2
(0, 0, 0) -> a^3
(0, 0, 1) -> a^2b
(0, 1, 1) -> ab^2
(1, 1, 1) -> b^3

Reply:
I understood your code. It is possible to compute the polynomial features without using 'ArrayFeatureExtractor' but I'm not sure it would do it in the same order than scikit-learn. I looked into the code of scikit-learn and the code is not really efficient too.